app/client/websocket/index.ts

```diff
@@ -3,218 +3,302 @@
-            }
+        if (this.connectingCallback) {
+            this.connectingCallback();
+        }
+
+        const regex = /^(?:https?|wss?):(?:\/\/)?[^/]*/;
+        const captured = (regex).exec(connectionUrl);
 
-            if (this.connectingCallback) {
-                this.connectingCallback();
+        let origin;
+        if (captured) {
+            origin = captured[0];
+
+            if (Platform.OS === 'android') {
+                // this is done cause for android having the port 80 or 443 will fail the connection
+                // the websocket will append them
+                const split = origin.split(':');
+                const port = split[2];
+                if (port === '80' || port === '443') {
+                    origin = `${split[0]}:${split[1]}`;
+                }
             }
+        } else {
+            // If we're unable to set the origin header, the websocket won't connect, but the URL is likely malformed anyway
+            const errorMessage = 'websocket failed to parse origin from ' + connectionUrl;
+            console.warn(errorMessage); // eslint-disable-line no-console
+            return;
+        }
 
-            const regex = /^(?:https?|wss?):(?:\/\/)?[^/]*/;
-            const captured = (regex).exec(connectionUrl);
+        let url = connectionUrl;
 
-            let origin;
-            if (captured) {
-                origin = captured[0];
+        const reliableWebSockets = system.config.EnableReliableWebSockets === 'true';
+        if (reliableWebSockets) {
+            // Add connection id, and last_sequence_number to the query param.
+            // We cannot also send it as part of the auth_challenge, because the session cookie is already sent with the request.
+            url = `${connectionUrl}?connection_id=${this.connectionId}&sequence_number=${this.serverSequence}`;
+        }
 
-                if (Platform.OS === 'android') {
-                    // this is done cause for android having the port 80 or 443 will fail the connection
-                    // the websocket will append them
-                    const split = origin.split(':');
-                    const port = split[2];
-                    if (port === '80' || port === '443') {
-                        origin = `${split[0]}:${split[1]}`;
-                    }
-                }
-            } else {
-                // If we're unable to set the origin header, the websocket won't connect, but the URL is likely malformed anyway
-                const errorMessage = 'websocket failed to parse origin from ' + connectionUrl;
-                console.warn(errorMessage); // eslint-disable-line no-console
-                reject(new Error(errorMessage));
-                return;
+        if (this.connectFailCount === 0) {
+            console.log('websocket connecting to ' + url); //eslint-disable-line no-console
+        }
+
+        this.conn = new WebSocket(url, [], {headers: {origin}});
```

Comment:
Here you should be using the WS from the `@mattermost/react-native-network-library`